### ISSUE TITLE
Datadog headers not added to requests made with Dio on web

### ISSUE THREAD

**Issue by a-h-mzd** [DataDog/dd-sdk-flutter#663]
### Question

When following the configuration steps in the documentation, we are instructed to use the enableHttpTracking function from the datadog_tracking_http_client package to track HTTP requests and add the necessary Datadog tracking headers. Based on my review of the code, it appears that this function overrides HttpOverrides.global with a custom implementation. I can confirm that requests are indeed being tracked, as they appear in the Datadog portal. However, Datadog-related headers are not being attached to these requests.

Please note that this issue is specific to web, and I haven’t tested it on other platforms. I suspect this approach might work on other platforms, as dio for native uses HttpOverrides to create the client for making requests. However, for browser environments, dio uses XMLHttpRequest directly for requests, which could be the reason headers aren't being attached.

**Comment by fuzzybinary** [comment]
Hi @a-h-mzd, thanks for the issue!

If requests are appearing in the portal, than the Browser SDK is intercepting them properly. Can you double check that you've properly setup `firstPartyHosts` properly, as this is the configuration option that tells Datadog which requests to attach tracing headers to.

**Comment by a-h-mzd** [comment]
It's mentioned in the documentation that `firstPartyHosts` must look like this:
```dart
final configuration = DatadogConfiguration(
  // configuration
  firstPartyHosts: ['example.com'],
)..enableHttpTracking()
```
And that is exactly how my current setup looks like.

**Comment by mathieux51** [comment]
@fuzzybinary for context [here](https://help.datadoghq.com/hc/en-us/requests/1889210) is the request that was made to support.

**Comment by fuzzybinary** [comment]
Hi @mathieux51 ,

Your support request had all the right information, thanks! It turns out there are a few issue in distributed tracing in Flutter Web, but Dio has nothing to do with it. 

First, the regex we supply to Web is made to look for the host name only, but the Browser SDK checks against the full URL.
Second, we never send the `traceSampleRate` option to Web.

I'm working on those issues now and adding a set of integration tests for tracing on Flutter web.

**Comment by mathieux51** [comment]
@fuzzybinary amazing news 🎉 . I'm very happy you're working on this! Thanks for your help 🙏

**Comment by a-h-mzd** [comment]
@fuzzybinary, I just tested #664 using our project, and it is working perfectly fine now. The traces are appearing in the portal as expected. Thank you for the fast and great support!

**Comment by a-h-mzd** [comment]
@fuzzybinary, Is there an ETA on when the new version with a fix for this issue will be released?

**Comment by fuzzybinary** [comment]
Hi @a-h-mzd,

Trying for this week. Ran into a bit of a roadblock yesturday but hopefully that should be worked out.

**Comment by fuzzybinary** [comment]
This should now be available in 2.9.0
